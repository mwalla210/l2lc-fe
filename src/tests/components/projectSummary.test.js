import React from 'react'
import renderer from 'react-test-renderer'
import ProjectSummary from '../../components/projectSummary'
jest.mock('../../components/barcode')
jest.mock('../../components/deleteModal')
jest.mock('../../components/promptModal')
jest.mock('../../components/buttonDefault')
jest.mock('reactstrap', () => ({
  ButtonGroup: 'ButtonGroup',
  DropdownItem: 'DropdownItem',
  DropdownMenu: 'DropdownMenu',
  ButtonDropdown: 'ButtonDropdown',
  DropdownToggle: 'DropdownToggle',
}))

const defaultOptions = {
  page: {
    summaryModel: {
      deleteModal: {
        openModal: jest.fn(),
        confirmAndClose: jest.fn(),
        modalOpen: false,
        closeModal: jest.fn(),
      },
      completeModal: {
        openModal: jest.fn(),
        confirmAndClose: jest.fn(),
        modalOpen: false,
        closeModal: jest.fn(),
      },
    },
    projectEditPage: jest.fn(),
    changeCustomerPage: jest.fn(),
  },
  website: {
    currentProject: {
      title: 'title',
      status: 'Completed',
      priority: '1-2 Days',
      costCenterTitle: 'CC',
      jobTypeTitle: 'JT',
      customer: {
        companyName: 'company',
        id: 1
      },
      timeSpent: 'time',
      partCount: 'parts',
      descr: 'descr',
      refNum: 'ref',
      barcodeDomID: 'barcodeDomID',
      barcodeScanID: 'barcodeScanID',
      hold: {
        flag: false
      },
      dateCreated: 'dateCreated',
      dateFinished: 'dateFinished'
    },
    currentUser: {
      admin: true
    },
    summaryMoreDropdownOpen: false,
    toggleSummaryMoreDD: jest.fn(),
    summaryActionsDropdownOpen: false,
    toggleSummaryActionsDD: jest.fn(),
  }
}

describe('ProjectSummary', () => {
  it ('Renders with snapshot (website.currentProject: priority 1-2 Days, status Completed, hold.flag false)', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <ProjectSummary {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders with snapshot (website.currentProject: priority 3 Days, status On Hold, hold.flag true)', () => {
    let options = Object.assign({}, defaultOptions)
    options.website.currentProject.priority = '3 Days'
    options.website.currentProject.status = 'On Hold'
    options.website.currentProject.hold.flag = true
    const component = renderer.create(
      <ProjectSummary {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders with snapshot (website.currentProject: priority 4-5 Days, status (!Completed, !On Hold), hold.flag false)', () => {
    let options = Object.assign({}, defaultOptions)
    options.website.currentProject.priority = '4-5 Days'
    options.website.currentProject.status = 'Received'
    const component = renderer.create(
      <ProjectSummary {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders and calls tasksClick', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <ProjectSummary {...options}/>,
    )
    const inst = component.getInstance()
    inst.tasksClick()
  })
  it ('Renders and calls timeEntries', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <ProjectSummary {...options}/>,
    )
    const inst = component.getInstance()
    inst.timeEntries()
  })
  it ('Renders and calls printClick', () => {
    let options = Object.assign({}, defaultOptions)
    global.print = jest.fn()
    const component = renderer.create(
      <ProjectSummary {...options}/>,
    )
    const inst = component.getInstance()
    inst.printClick()
    expect(global.print).toBeCalled()
  })
})
