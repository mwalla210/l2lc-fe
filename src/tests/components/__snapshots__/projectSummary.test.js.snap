// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectSummary Renders with snapshot (website.currentProject: priority 1-2 Days, status Completed, hold.flag false) 1`] = `
<div>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-3"
      style={
        Object {
          "border": "1px solid #b4b4b4",
          "borderRadius": "8px",
          "margin": "2px",
          "paddingTop": "1rem",
        }
      }
    >
      <h5>
        Main Information
      </h5>
      <div>
        <h6>
          Description
        </h6>
        <p>
          title
        </p>
      </div>
      <div>
        <h6>
          Priority
        </h6>
        <p>
          <span>
            <img
              src="../../style/open-iconic-master/svg/clock.svg"
              style={
                Object {
                  "width": "13px",
                }
              }
            />
             1-2 Days
          </span>
        </p>
      </div>
      <div>
        <h6>
          Status
        </h6>
        <p>
          <span>
            <span
              style={
                Object {
                  "color": "#49a4ff",
                  "transition": "all .3s ease",
                }
              }
            >
              ●
            </span>
             Completed
          </span>
        </p>
      </div>
      <div>
        <h6>
          Cost Center
        </h6>
        <p>
          CC
        </p>
      </div>
      <div>
        <h6>
          Project Type
        </h6>
        <p>
          JT
        </p>
      </div>
    </div>
    <div
      className="col-3"
      style={
        Object {
          "border": "1px solid #b4b4b4",
          "borderRadius": "8px",
          "margin": "2px",
          "paddingTop": "1rem",
        }
      }
    >
      <h5>
        Additional Information
      </h5>
      <div>
        <h6>
          Customer Name
        </h6>
        <p>
          company
        </p>
      </div>
      <div>
        <h6>
          Date Created
        </h6>
        <p>
          dateCreated
        </p>
      </div>
      <div>
        <h6>
          Date Finished
        </h6>
        <p>
          dateFinished
        </p>
      </div>
      <div>
        <h6>
          Time Spent
        </h6>
        <p>
          time
        </p>
      </div>
      <div>
        <h6>
          Part Count
        </h6>
        <p>
          parts
        </p>
      </div>
      <div>
        <h6>
          Details
        </h6>
        <p>
          descr
        </p>
      </div>
      <div>
        <h6>
          Job/Work Order Number
        </h6>
        <p>
          ref
        </p>
      </div>
    </div>
  </div>
  <div
    className="row justify-content-center"
  />
  <div
    className="row justify-content-center"
  >
    <ButtonGroup>
      <ButtonDropdown
        isOpen={false}
        toggle={[MockFunction]}
      >
        <DropdownToggle
          caret={true}
          color="secondary"
          outline={true}
        >
          More...
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem
            onClick={[Function]}
          >
            Tasks
          </DropdownItem>
          <DropdownItem
            onClick={[Function]}
          >
            Time Entries
          </DropdownItem>
          <DropdownItem
            onClick={[Function]}
          >
            Customer
          </DropdownItem>
        </DropdownMenu>
      </ButtonDropdown>
      <ButtonDropdown
        isOpen={false}
        toggle={[MockFunction]}
      >
        <DropdownToggle
          caret={true}
          color="primary"
        >
          Actions
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem
            disabled={true}
            onClick={[MockFunction]}
          >
            Edit Details
          </DropdownItem>
          <DropdownItem
            disabled={true}
            onClick={[MockFunction]}
          >
            Edit Customer
          </DropdownItem>
          <DropdownItem
            disabled={true}
            onClick={[MockFunction]}
          >
            Complete
          </DropdownItem>
          <DropdownItem
            onClick={[MockFunction]}
          >
            Delete
          </DropdownItem>
        </DropdownMenu>
      </ButtonDropdown>
    </ButtonGroup>
  </div>
</div>
`;

exports[`ProjectSummary Renders with snapshot (website.currentProject: priority 3 Days, status On Hold, hold.flag true) 1`] = `
<div>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-3"
      style={
        Object {
          "border": "1px solid #b4b4b4",
          "borderRadius": "8px",
          "margin": "2px",
          "paddingTop": "1rem",
        }
      }
    >
      <h5>
        Main Information
      </h5>
      <div>
        <h6>
          Description
        </h6>
        <p>
          title
        </p>
      </div>
      <div>
        <h6>
          Priority
        </h6>
        <p>
          <span>
            <img
              src="../../style/open-iconic-master/svg/calendar.svg"
              style={
                Object {
                  "width": "13px",
                }
              }
            />
             3 Days
          </span>
        </p>
      </div>
      <div>
        <h6>
          Status
        </h6>
        <p>
          <span>
            <span
              style={
                Object {
                  "color": "#ffbf00",
                  "transition": "all .3s ease",
                }
              }
            >
              ●
            </span>
             On Hold
          </span>
        </p>
      </div>
      <div>
        <h6>
          Cost Center
        </h6>
        <p>
          CC
        </p>
      </div>
      <div>
        <h6>
          Project Type
        </h6>
        <p>
          JT
        </p>
      </div>
    </div>
    <div
      className="col-3"
      style={
        Object {
          "border": "1px solid #b4b4b4",
          "borderRadius": "8px",
          "margin": "2px",
          "paddingTop": "1rem",
        }
      }
    >
      <h5>
        Additional Information
      </h5>
      <div>
        <h6>
          Customer Name
        </h6>
        <p>
          company
        </p>
      </div>
      <div>
        <h6>
          Date Created
        </h6>
        <p>
          dateCreated
        </p>
      </div>
      <div>
        <h6>
          Date Finished
        </h6>
        <p>
          dateFinished
        </p>
      </div>
      <div>
        <h6>
          Time Spent
        </h6>
        <p>
          time
        </p>
      </div>
      <div>
        <h6>
          Part Count
        </h6>
        <p>
          parts
        </p>
      </div>
      <div>
        <h6>
          Details
        </h6>
        <p>
          descr
        </p>
      </div>
      <div>
        <h6>
          Job/Work Order Number
        </h6>
        <p>
          ref
        </p>
      </div>
    </div>
  </div>
  <div
    className="row justify-content-center"
  />
  <div
    className="row justify-content-center"
  >
    <ButtonGroup>
      <ButtonDropdown
        isOpen={false}
        toggle={[MockFunction]}
      >
        <DropdownToggle
          caret={true}
          color="secondary"
          outline={true}
        >
          More...
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem
            onClick={[Function]}
          >
            Tasks
          </DropdownItem>
          <DropdownItem
            onClick={[Function]}
          >
            Time Entries
          </DropdownItem>
          <DropdownItem
            onClick={[Function]}
          >
            Customer
          </DropdownItem>
        </DropdownMenu>
      </ButtonDropdown>
      <ButtonDropdown
        isOpen={false}
        toggle={[MockFunction]}
      >
        <DropdownToggle
          caret={true}
          color="primary"
        >
          Actions
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem
            disabled={false}
            onClick={[MockFunction]}
          >
            Edit Details
          </DropdownItem>
          <DropdownItem
            disabled={false}
            onClick={[MockFunction]}
          >
            Edit Customer
          </DropdownItem>
          <DropdownItem
            disabled={false}
            onClick={[MockFunction]}
          >
            Complete
          </DropdownItem>
          <DropdownItem
            onClick={[MockFunction]}
          >
            Delete
          </DropdownItem>
        </DropdownMenu>
      </ButtonDropdown>
    </ButtonGroup>
  </div>
</div>
`;

exports[`ProjectSummary Renders with snapshot (website.currentProject: priority 4-5 Days, status (!Completed, !On Hold), hold.flag false) 1`] = `
<div>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-3"
      style={
        Object {
          "border": "1px solid #b4b4b4",
          "borderRadius": "8px",
          "margin": "2px",
          "paddingTop": "1rem",
        }
      }
    >
      <h5>
        Main Information
      </h5>
      <div>
        <h6>
          Description
        </h6>
        <p>
          title
        </p>
      </div>
      <div>
        <h6>
          Priority
        </h6>
        <p>
          <span>
            <img
              src="../../style/open-iconic-master/svg/calendar.svg"
              style={
                Object {
                  "width": "13px",
                }
              }
            />
             4-5 Days
          </span>
        </p>
      </div>
      <div>
        <h6>
          Status
        </h6>
        <p>
          <span>
            <span
              style={
                Object {
                  "color": "#57d500",
                  "transition": "all .3s ease",
                }
              }
            >
              ●
            </span>
             Received
          </span>
        </p>
      </div>
      <div>
        <h6>
          Cost Center
        </h6>
        <p>
          CC
        </p>
      </div>
      <div>
        <h6>
          Project Type
        </h6>
        <p>
          JT
        </p>
      </div>
    </div>
    <div
      className="col-3"
      style={
        Object {
          "border": "1px solid #b4b4b4",
          "borderRadius": "8px",
          "margin": "2px",
          "paddingTop": "1rem",
        }
      }
    >
      <h5>
        Additional Information
      </h5>
      <div>
        <h6>
          Customer Name
        </h6>
        <p>
          company
        </p>
      </div>
      <div>
        <h6>
          Date Created
        </h6>
        <p>
          dateCreated
        </p>
      </div>
      <div>
        <h6>
          Date Finished
        </h6>
        <p>
          dateFinished
        </p>
      </div>
      <div>
        <h6>
          Time Spent
        </h6>
        <p>
          time
        </p>
      </div>
      <div>
        <h6>
          Part Count
        </h6>
        <p>
          parts
        </p>
      </div>
      <div>
        <h6>
          Details
        </h6>
        <p>
          descr
        </p>
      </div>
      <div>
        <h6>
          Job/Work Order Number
        </h6>
        <p>
          ref
        </p>
      </div>
    </div>
  </div>
  <div
    className="row justify-content-center"
  />
  <div
    className="row justify-content-center"
  >
    <ButtonGroup>
      <ButtonDropdown
        isOpen={false}
        toggle={[MockFunction]}
      >
        <DropdownToggle
          caret={true}
          color="secondary"
          outline={true}
        >
          More...
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem
            onClick={[Function]}
          >
            Tasks
          </DropdownItem>
          <DropdownItem
            onClick={[Function]}
          >
            Time Entries
          </DropdownItem>
          <DropdownItem
            onClick={[Function]}
          >
            Customer
          </DropdownItem>
        </DropdownMenu>
      </ButtonDropdown>
      <ButtonDropdown
        isOpen={false}
        toggle={[MockFunction]}
      >
        <DropdownToggle
          caret={true}
          color="primary"
        >
          Actions
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem
            disabled={false}
            onClick={[MockFunction]}
          >
            Edit Details
          </DropdownItem>
          <DropdownItem
            disabled={false}
            onClick={[MockFunction]}
          >
            Edit Customer
          </DropdownItem>
          <DropdownItem
            disabled={false}
            onClick={[MockFunction]}
          >
            Complete
          </DropdownItem>
          <DropdownItem
            onClick={[MockFunction]}
          >
            Delete
          </DropdownItem>
        </DropdownMenu>
      </ButtonDropdown>
    </ButtonGroup>
  </div>
</div>
`;
